File: 03-poo/02-atributos-metodos-y-estado/scala/soluciones/RetoContador.scala
class ContadorReto:
  private var valor = 0
  def incrementar(): Unit = valor += 1
  def mostrarValor: Int = valor

@main def retoContador(): Unit =
  val c = new ContadorReto
  c.incrementar(); c.incrementar()
  println(c.mostrarValor)

File: 03-poo/02-atributos-metodos-y-estado/java/soluciones/RetoContador.java
public class RetoContador {
    static class Contador {
        private int valor;
        void incrementar() { valor++; }
        int mostrarValor() { return valor; }
    }
    public static void main(String[] args) {
        Contador contador = new Contador();
        contador.incrementar();
        contador.incrementar();
        System.out.println(contador.mostrarValor());
    }
}
